lib/ecto/adapters/tds.ex

```diff
@@ -295,4 +295,11 @@ defmodule Ecto.Adapters.Tds do
 
     result
   end
+
+  defp log_options(opts) do
+    case Keyword.get(opts, :log_sql_mode, "commands") do
+      "all" -> [log_sql: Keyword.get(opts, :log_sql, :info), log: Keyword.get(opts, :log, :info)]
+      _ -> [log: false]
+    end
+  end
```

Comment:
Couldn't this be managed in advance by the mix task so this logic doesn't get "leaked" into adapters?

Reply:
:+1: The adapters could just handle log_sql and log directly as log levels. :)

Reply:
:+1: Make `:log_sql_mode` to be a property of the Mix task only. From there on, it is all log_sql and log which are either false or a log level.

Reply:
I wish it were that simple 😅  if I set `log` and `log_sql` at the mix tasks, then I only have 2 flags I can pass down. There are several combinations:

1. Log nothing (eg, `--quiet`) `log: false, log_sql: false`
2. Log changes `log: true, log_sql: false`
3. Log SQL from the changes `log: true, log_sql: true`
4. NEW:  Log the commands and the locking/version/transaction commands. There's not a way that I see to do that without adding another option (currently, by leaking `log_sql_mode` to adapters and setting the two log options)

Since locking/transactions is handled in the adapters, that's why I had to leak `log_sql_mode` down there. SchemaMigration is the other place so it could log the insert/delete of the version.

Any suggestions?

Reply:
What I imagine is doing, in the edge, something like (but more beautiful 😄 ):

```elixir
opts = Keyword.put(opts, :log_all, opts[:log] && opts[:log_sql_mode] == "all" && opts[:log_sql])
 ```

and later doing something as:

```elixir
# for internal stuff
blah(..., Keyword.merge(opts, [log: opts[:log_all], timeout: :infinity])

# for the rest:
blah(..., opts)
```

Reply:
Yeah, I think I understand now. I'll set the `log_all: true|false` key in the mix task, separating it from the `--log-sql-mode` switch, and let each of the adapters + SchemaMigration look for that key to determine when `log: true` for those lock/version/transaction commands.

Reply:
Should be ready for review again. Thanks for the suggestion.